### ISSUE TITLE
AGC and mixed precision not compatible?

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#19]
When training a 3D U-Net like architecture, with adaptive gradient clipping and mixed precision enabled, it was observed that the loss was increasing instead of decreasing.

The network trained fine if AGC was enabled and mixed precision was not, or vica versa. Hence, these two implementations does not seem to be compatible.

**Comment by andreped** [comment]
We should use AGC on unscaled gradients. Hence, the order was wrong.

This is at least how it is done in pytorch:
https://pytorch.org/docs/stable/notes/amp_examples.html#gradient-clipping

**Comment by andreped** [comment]
Added fix for the ordering: cbf6b5ad5b0d58a62a73252ebeb51f9f13e396dd

Test if it works by removing the old version, and reinstalling from source, @dbouget:
```
pip uninstall gradient-accumulator
pip install git+https://github.com/andreped/GradientAccumulator
```

Note that simply installing from source might not overwrite local, as the version has not changed. A forced uninstall before is probably a good idea.

**Comment by dbouget** [comment]
You have the train loss top and val loss bottom
![image](https://user-images.githubusercontent.com/24223628/175564577-53fa88df-c668-4fd6-a833-411632b7a26d.png)

All combinations seem to converge with the fix for the ordering, might just have to start from a lower initial lr with agc (used 1e-4).
